### ISSUE TITLE
State of Chihaya? No release since 2016? Will a new release be published?

### ISSUE THREAD

**Issue by polarathene** [chihaya/chihaya#547]
I had a quick look via Github and noticed the following:

- June 2017 - [Last tagged release](https://github.com/chihaya/chihaya/tags) is `v1.0.2` in Oct 2016, and `v2.0.0-rc.2` in June 2017 (`rc.1` prior in Oct 2016).
- Nov 2017 - [Stating the 2.0 release is nearly ready](https://github.com/chihaya/chihaya/issues/202#issuecomment-347028255).
- Feb 2019 - [The last remaining 2.0 milestone issue](https://github.com/chihaya/chihaya/issues/396#issuecomment-419715402) (still open) had comment activity in Sep 2018; and was added to the 2.0 milestone in Feb 2019.
- April 2019 - The [container badge](https://quay.io/repository/jzelinskie/chihaya) links to Quay which notes the last activity as April 2019.
- Aug 2021 - The project README makes no mention of any DockerHub or GHCR (Github registry), and no other obvious mentions of current version beyond Github Releases and a README disclaimer about the master branch advising to use Github Releases (v1.0.2 from Oct 2016).

---

I have only browsed through Github UI for commit history back to around Feb 2019, where any actual development seems to have last occurred. Bulk of commits since seem to mostly be minor fixes, some docs + CI/maintenance additions, and then automated dependency updates dominate.

I noticed that there was a `refactor` branch mentioned](https://github.com/chihaya/chihaya/issues/202#issuecomment-243512320) in Aug 2016 by an active maintainer of the project @jzelinskie , that branch doesn't exist anymore, perhaps it was merged but given the above findings, it's unclear what the state of `chihaya` is?

I see that in 2021 there is some more development activity in open PRs, [such as this one](https://github.com/chihaya/chihaya/pull/540) from July which is still active. Is development picking up again?

---

## TL;DR

Is the advice to continue using `chihaya` release from Github Releases v1.0.2 from Oct 2016, or the Docker image on Quay from April 2019, is there any notable differences from that much time apart? (_I haven't yet looked into the container build, perhaps it builds/pulls the same v1.0.2 release thus no real difference?_)

Wouldn't it make sense to make a new release after all these years with the fixes and especially the dependency updates?

Development picking up pace again is great, but I'm a little concerned it could continue to delay a new release being pushed out..?

**Comment by jzelinskie** [comment]
We honestly just haven't been aggressively doing releases. Most chihaya users are running HEAD rather than using stable releases. If you'd like to use docker, I recommend https://quay.io/repository/jzelinskie/chihaya-git?tag=latest&tab=tags

If we automated our release process with something like goreleaser, it might help communicate the best practice.

**Comment by pataquets** [comment]
To those not finding this issue, truth is, looks a bit confusing.
The "stable" public Docker image even doesn't have a `:latest` tag.
Adding some notice or even a link to this issue in the Readme would help users, IMO.

**Comment by polarathene** [comment]
> The "stable" public Docker image even doesn't have a `:latest` tag.

Are you referring to the quay.io link to `chihaya-git`? It has a `:latest` tag but as the image name implies it's meant to be the latest git build, not stable releases.

It doesn't seem like it gets regularly updated though (last update 7 months ago, Jan 2022), so you're probably better off [building from the repo `Dockerfile` directly](https://github.com/chihaya/chihaya/blob/main/Dockerfile) instead of pulling that image.

**EDIT:** Oh I see the README mentions the docker images and also references a [`stable` image that you're referring to](https://quay.io/repository/jzelinskie/chihaya?tab=tags&tag=latest), you're not really meant to rely on `:latest` tags for production deployment, so specifying the last known release tag is technically ok, just a bit inconvenient for testing.

> Adding some notice or at least a link to this issue in the Readme would help users, IMO.

Probably a good suggestion, but this issue is open, as long as that remains the case the maintainers could also pin this issue for visibility, since anyone checking out the README to get a feel for the project would probably skim through the issues of the project too.

A better improvement would be to just drop all mentions of stable releases, there's been no new releases at all since 2017 and maintainer advice is to build from git directly.. I can't imagine the maintainers are interested in users feedback from any of the tagged releases, any issues found have probably been addressed since.

Personally, if this is how users are choosing to run `chihaya` and it's been working well for so long, just add scheduled CI that tags releases (_eg: monthly `2022-07` / `22.7`, provided there has been new commits_).
